apps/admin-gui/src/app/shared/route-auth-guard.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivateChild,
+  Router,
+  RouterStateSnapshot,
+  UrlTree,
+} from '@angular/router';
+import { Observable } from 'rxjs';
+import {
+  GuiAuthResolver,
+  NotificatorService,
+  RoutePolicyService,
+} from '@perun-web-apps/perun/services';
+import { PerunBean } from '@perun-web-apps/perun/openapi';
+
+interface AuthPair {
+  key: string;
+  entity: PerunBean;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RouteAuthGuardService implements CanActivateChild {
+  constructor(
+    private authResolver: GuiAuthResolver,
+    private routePolicyService: RoutePolicyService,
+    private router: Router,
+    private notificator: NotificatorService
+  ) {}
+
+  private static getBeanName(key: string): string {
+    switch (key) {
+      case 'o':
+        return 'Vo';
+      case 'g':
+        return 'Group';
+      case 'f':
+        return 'Facility';
+      case 'r':
+        return 'Resource';
```

Comment:
ok, I guess this is the best way
but wouldn't be better in that case that there would be whole names of the path, like "organizations" instead of "o"
its better code readability and also when we one day change the URL path in can be easier to identify which pieces of code needs to be changed as well
btw members path is missing here